### PR TITLE
feat: add information to token toggle screen

### PR DESCRIPTION
**Pull request by reidzeibel**
closes https://github.com/AtB-AS/kundevendt/issues/8653

This PR will create a new component for token toggle information, and use that component on both screens referred on the issue as shown below.

<details>

<summary>Vis skjermbilde</summary>

image

</details>

I could just copy everything from the first screen to second screen and it would work with a little bit of styling but I thought code duplication wouldn't be a good thing, so I tried to make a new component. There is an optional `textColor` property that is added to the component to keep the color of the text stay visible on light mode (see below). Without specifying the `textColor`, it will turn to black text and wouldn't look good. I used the example on ScreenHeader.tsx to determine the color of the text that I should use, so it would look the same as the `Back` navigation button on the top left, which stays the same regardless of the color mode we use on the app.

<details>

<summary>Vis skjermbilde</summary>

image

</details>

I also moved the `use-token-toggle-details.tsx` to utils folder as it would make more sense and should be on the same place as the other react hooks.

### CODE REVIEW

**Review by gorandalum — CHANGES_REQUESTED** [review]
In general good work, just some comments on some React nuances and some other nuances about how we do it in the app. Let's go through them together at the office 😊

File: src/token-toggle-info/index.ts
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/index.ts:1]: Updated as suggestion.

File: src/select-travel-token-screen/SelectTravelTokenScreenComponent.tsx [hunk 1]
```diff
@@ -41,7 +44,15 @@ export const SelectTravelTokenScreenComponent = ({onAfterSave}: Props) => {
...
 
-  const {tokens, toggleToken} = useMobileTokenContextState();
+  const {tokens, deviceInspectionStatus, toggleToken} =
+    useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
+  const shouldFetchTokenDetails =
+    screenHasFocus && deviceInspectionStatus !== 'loading';
+  const {shouldShowLoader, toggleLimit} = useTokenToggleDetails(
+    shouldFetchTokenDetails,
+  );
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/select-travel-token-screen/SelectTravelTokenScreenComponent.tsx:54]: the toggleLimit is used in other screen as well so I will have to look deeper to find a solution for this.

File: src/token-toggle-info/TokenToggleInfoComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  const loader = Loader(
+    componentType === 'plain' ? style.loader : style.sectionedLoader,
+  );
+
+  const content = Content({
+    containerStyle:
+      componentType === 'plain' ? style.container : style.sectionedContainer,
+    toggleLimit: toggleLimit,
+    textColor: textColor,
+  });
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/TokenToggleInfoComponent.tsx]: Updated as suggestion.

File: src/token-toggle-info/TokenToggleInfoComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  formatToShortDateWithYear,
+  formatToVerboseFullDate,
+} from '@atb/utils/date';
+import {StyleSheet} from '@atb/theme';
+import {StaticColor} from '@atb/theme/colors';
+
+type ComponentProps = {
+  toggleLimit: number;
+  shouldShowLoader?: boolean;
+  componentType: 'plain' | 'sectioned';
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/TokenToggleInfoComponent.tsx]: Updated as suggestion.

File: src/token-toggle-info/TokenToggleInfoComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  });
+
+  return shouldShowLoader ? loader : content;
+};
+
+const useStyles = StyleSheet.createThemeHook((theme) => ({
+  container: {
+    flexDirection: 'row',
+    marginTop: theme.spacings.small,
+    marginBottom: theme.spacings.large,
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/TokenToggleInfoComponent.tsx:39]: Updated as suggestion.

File: src/token-toggle-info/TokenToggleInfoComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  sectionedContainer: {
+    flexDirection: 'row',
+  },
+  sectionedLoader: {
+    alignSelf: 'center',
+    flex: 1,
+  },
+}));
+
+const Loader = (style?: ViewStyle) => <ActivityIndicator style={style} />;
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/TokenToggleInfoComponent.tsx]: Updated as suggestion.

File: src/token-toggle-info/TokenToggleInfoComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        );
+      default:
+        return t(
+          TravelTokenTexts.travelToken.toggleCountLimitInfo(
+            toggleLimit,
+            countRenewalDate,
+          ),
+        );
+    }
+  };
```
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/token-toggle-info/TokenToggleInfoComponent.tsx]: Updated as suggestion.

File: src/select-travel-token-screen/SelectTravelTokenScreenComponent.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (reidzeibel) [reply src/select-travel-token-screen/SelectTravelTokenScreenComponent.tsx]: Ok found it, ready for review

**Pull request merged**